File: templates/starter-kits/python-project/scripts/validate-governance.py
#!/usr/bin/env python3
"""
Governance Validation Script

Validates that the project meets governance requirements.
"""

import os
import sys
from pathlib import Path

REQUIRED_FILES = [
    '.repo/repo.manifest.yaml',
    '.repo/README.md',
    'pyproject.toml',
    'README.md',
    'src/__init__.py',
]

def main():
    """Main validation function"""
    errors = 0
    
    print("🔍 Validating governance compliance...\n")
    
    print("Checking required files...")
    for file in REQUIRED_FILES:
        if Path(file).exists():
            print(f"  ✓ {file}")
        else:
            print(f"  ✗ {file} - MISSING")
            errors += 1
    
    print("\n" + "=" * 50)
    if errors == 0:
        print("✅ All governance checks passed!")
        sys.exit(0)
    else:
        print(f"❌ {errors} governance check(s) failed!")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: templates/starter-kits/javascript-nodejs/scripts/validate-governance.js
#!/usr/bin/env node

/**
 * Governance Validation Script
 * 
 * Validates that the project meets governance requirements
 */

const fs = require('fs');
const path = require('path');

const REQUIRED_FILES = [
  '.repo/repo.manifest.yaml',
  '.repo/README.md',
  'package.json',
  'README.md'
];

const REQUIRED_SCRIPTS = [
  'test',
  'lint',
  'governance:validate'
];

let errors = 0;

console.log('🔍 Validating governance compliance...\n');

// Check required files exist
console.log('Checking required files...');
REQUIRED_FILES.forEach(file => {
  const filePath = path.join(process.cwd(), file);
  if (fs.existsSync(filePath)) {
    console.log(`  ✓ ${file}`);
  } else {
    console.log(`  ✗ ${file} - MISSING`);
    errors++;
  }
});

// Check package.json has required scripts
console.log('\nChecking required scripts...');
const packageJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
REQUIRED_SCRIPTS.forEach(script => {
  if (packageJson.scripts && packageJson.scripts[script]) {
    console.log(`  ✓ ${script}`);
  } else {
    console.log(`  ✗ ${script} - MISSING`);
    errors++;
  }
});

// Report results
console.log('\n' + '='.repeat(50));
if (errors === 0) {
  console.log('✅ All governance checks passed!');
  process.exit(0);
} else {
  console.log(`❌ ${errors} governance check(s) failed!`);
  process.exit(1);
}
